### PR TITLE
Development guidelines

### PR DESCRIPTION
**Pull request by adriacidre**
I've been reading some recommendations about how status-go should be organized. Even if these recommendations have been already discussed on slack or github at some point, they're not documented anywhere. This can make it difficult for everybody on the team to follow.

Basically this doc aims to group all general guidelines to have in mind when you work with status-go.

Please feel free to comment with your own ideas and i'll add them to the doc.

### CODE REVIEW

**Comment by themue** [comment]
Add a link to https://wiki.status.im/Go_Code_Quality_Notes.

<blockquote><div><strong><a href="https://wiki.status.im/Go_Code_Quality_Notes">Go Code Quality Notes - Status Wiki</a></strong></div></blockquote>

File: _assets/docs/DEVELOPMENT_GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+
+### /geth/ path
+
+Some status-go code lives inside `/geth` path, however this path no longer makes sense and we should get rid of it. Please don't store your new packages on `/geth` use the root path of the repo instead.
+
+
+### Server as standalone app
+
+`status-go` is used both for mobile devices and also on our nodes. We should try to split this in order to not overload single repository.
```
Comment (divan) [inline _assets/docs/DEVELOPMENT_GUIDELINES.md:11]: There is almost no overload from my POV. We have most of the codebase shared, and only two  small "entrance" packages (`cmd/statusd` and `lib`) for server and mobile respectively.
I'm not sure what problem splitting this into two repositories will solve.

Comment (adriacidre) [reply _assets/docs/DEVELOPMENT_GUIDELINES.md:11]: discussion on this topic comes from [this issue](https://github.com/status-im/status-go/issues/846)

Comment (dshulyak) [reply _assets/docs/DEVELOPMENT_GUIDELINES.md:11]: - signals are completely ignored when statusd is used
- rpc interceptor doesn't make any sense for statusd
- shhext is not necessary with statusd
- different sources of defaults for config files and statusd

i am pretty sure there are more things

File: _assets/docs/DEVELOPMENT_GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+This document aims to group all general guidelines to have in mind when you work with status-go.
+
+
+### /geth/ path
```
Comment (dshulyak) [inline _assets/docs/DEVELOPMENT_GUIDELINES.md:4]: does it make sense to create this guideline? maybe just move all code from geth namespace?

File: _assets/docs/DEVELOPMENT_GUIDELINES.md (hunk 1 above)
Comment (dshulyak) [reply _assets/docs/DEVELOPMENT_GUIDELINES.md:11]: also this must be tracked as a simple issue on github, rather then guideline

**Review by dshulyak — CHANGES_REQUESTED** [review]
i am not sure if these things worse adding as guidelines. can we create 2 github issues to make these things happen? Even if 2nd is controversial - we can have a discussion in the github issue

File: _assets/docs/DEVELOPMENT_GUIDELINES.md (hunk 2 above)
Comment (adriacidre) [reply _assets/docs/DEVELOPMENT_GUIDELINES.md:4]: 👍  i can create an issue instead

Comment (adriacidre) [reply _assets/docs/DEVELOPMENT_GUIDELINES.md:4]: Issue on moving `geth` package to repo root : https://github.com/status-im/status-go/issues/856

**Comment by adriacidre** [comment]
Cool, closing this PR as main points of this doc will be addressed with a different approach :)

**Pull request closed without merging**